app/src/main/kotlin/org/akanework/gramophone/logic/GramophoneExtensions.kt

```diff
@@ -27,6 +27,7 @@ import android.content.res.Configuration
 import android.graphics.Color
 import android.graphics.drawable.AnimatedVectorDrawable
 import android.graphics.drawable.Drawable
+import android.media.MediaMetadataRetriever
```

Comment:
Can you use https://developer.android.com/reference/androidx/media3/exoplayer/MetadataRetriever#retrieveMetadata(androidx.media3.exoplayer.source.MediaSource.Factory,androidx.media3.common.MediaItem) instead? ExoPlayer is more predictable as it's shipped as part of app code.

Reply:
After researching, I found that `MetadataRetriever.retrieveMetadata()` does not provide bitrate information for local files; it primarily retrieves embedded metadata like ID3 tags. To obtain the actual bitrate, ExoPlayer must be running and playing the media. On the other hand, `MediaMetadataRetriever` works for local files without requiring playback.